### PR TITLE
A new transformer added to the latest version of the develop branch

### PR DESCRIPTION
**Pull request by leila-rashidi**

### CODE REVIEW

**Comment by leila-rashidi** [comment]
Hi @subbyte and @svajipay, I just changed one file. Hope that we pass the code style compliance check.

**Comment by subbyte** [comment]
We follow the [black style](https://github.com/opencybersecurityalliance/kestrel-lang/blob/develop/CONTRIBUTING.rst#code-style) for Kestrel code. Looks like you already figure it out. Let me rerun the checks.

**Comment by leila-rashidi** [comment]
> We follow the [black style](https://github.com/opencybersecurityalliance/kestrel-lang/blob/develop/CONTRIBUTING.rst#code-style) for Kestrel code. Looks like you already figure it out. Let me rerun the checks.

Thank you @subbyte . I reformatted.

**Comment by subbyte** [comment]
@leila-rashidi Could you undo the last commit 83c51190bed3cef75d7267c8374c84929575e9a4? It looks like you already fixed the styling issue in bfd79136913beb5ac5e067c5a24ec127a114ee07, but introduce lots of styling changes in `docs/` and `tests/` in 83c51190bed3cef75d7267c8374c84929575e9a4, which we intentionally don't do.

When you see a github action failed, you can go to the specific github workflow to see how to fix it, e.g., for the styling one, you can follow https://github.com/opencybersecurityalliance/kestrel-lang/blob/develop/.github/workflows/code-style.yml

**Comment by leila-rashidi** [comment]
@subbyte , I reverted the reformatting changes, and then reformatted two files which had styling issue.

File: src/kestrel/syntax/kestrel.lark [hunk 1]
```diff
@@ -76,11 +76,16 @@ expression: vtrans where_clause? attr_clause? sort_clause? limit_clause? offset_
 // the function in transformer will mal-function in `merge_transformers()`
 vtrans: TRANSFORM "(" VARIABLE ")"
       | VARIABLE
+      | TRANSFORM2 "(" VARIABLE ")"
```
Comment (pcoccoli) [inline src/kestrel/syntax/kestrel.lark]: I don't think we need TRANSFORM2.  Instead, you can "or" OBSERVED with TIMESTAMPED in TRANSFORM.

File: src/kestrel/syntax/kestrel.lark [hunk 2]
```diff
@@ -76,11 +76,16 @@ expression: vtrans where_clause? attr_clause? sort_clause? limit_clause? offset_
...
 
 TRANSFORM: (TIMESTAMPED)
 
 TIMESTAMPED: "TIMESTAMPED"i
 
+TRANSFORM2: (OBSERVED)
+
+OBSERVED: "OBSERVED"i
```
Comment (pcoccoli) [inline src/kestrel/syntax/kestrel.lark]: I don't know if OBSERVED is a good name here since it only adds the `id` attr and not any of the others.

Comment (leila-rashidi) [reply src/kestrel/syntax/kestrel.lark]: Hi @pcoccoli,

Thanks for your comment. I changed OBSERVED with ADDOBSID.

File: src/kestrel/syntax/kestrel.lark (hunk 1 above)
Comment (leila-rashidi) [reply src/kestrel/syntax/kestrel.lark:79]: Hi @pcoccoli , Unfortunately, using or raises error.

Comment (leila-rashidi) [reply src/kestrel/syntax/kestrel.lark]: Hello @pcoccoli , 

As per your request, I made additional changes to have only one rule to describe transformers.

Regards

**Pull request closed without merging**